### ISSUE TITLE
`Layout/SpaceInsideArrayPercentLiteral` and `Layout/SpaceInsidePercentLiteralDelimiters` are unaware of vertical alignment of `%w[]` arrays

### ISSUE THREAD

**Issue by postmodern** [rubocop/rubocop#11612]
Sometimes it is necessary to represent a table as an Array of Arrays of Strings.

```ruby
table = [
  %w[A   B   C  ],
  %w[foo bar baz]
]
```

However, the `Layout/SpaceInsideArrayPercentLiteral` and `Layout/SpaceInsidePercentLiteralDelimiters` do not seems to be aware of the vertical alignment of each value so that the values align with the previous/next rows values to form columns.

--------

## Expected behavior

No warnings or other unrelated warnings.

## Actual behavior

```
For /home/postmodern/test/ruby/rubocop: configuration from /home/postmodern/test/ruby/rubocop/.rubocop.yml
Default configuration from /home/postmodern/test/ruby/rubocop/vendor/bundle/ruby/3.1.0/gems/rubocop-1.45.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/postmodern/test/ruby/rubocop/test.rb
W

Offenses:

test.rb:3:1: W: Lint/UselessAssignment: Useless assignment to variable - table.
table = [
^^^^^
test.rb:4:7: C: [Correctable] Layout/SpaceInsideArrayPercentLiteral: Use only a single space inside array percent literal.
  %w[A   B   C  ],
      ^^^
test.rb:4:11: C: [Correctable] Layout/SpaceInsideArrayPercentLiteral: Use only a single space inside array percent literal.
  %w[A   B   C  ],
          ^^^
test.rb:4:15: C: [Correctable] Layout/SpaceInsidePercentLiteralDelimiters: Do not use spaces inside percent literal delimiters.
  %w[A   B   C  ],
              ^^

1 file inspected, 4 offenses detected, 3 offenses autocorrectable
Finished in 0.2911929000001692 seconds
```

## Steps to reproduce the problem

`Gemfile`:

```ruby
source 'https://rubygems.org'

gem 'rubocop'
```

`test.rb`:

```ruby
# frozen_string_literal: true

table = [
  %w[A   B   C  ],
  %w[foo bar baz]
]
```

1. `bundle install`
2. `bundle exec robucop --debug test.rb`

## RuboCop version

```
$ bundle exec rubocop -V
1.45.1 (using Parser 3.2.1.0, rubocop-ast 1.26.0, running on ruby 3.1.3) [x86_64-linux]
```
